web/src/reactions.js

```diff
@@ -468,44 +468,73 @@ export function set_clean_reactions(message) {
     // to make sure reify_message_id moves the data structure
     // properly.
     message.clean_reactions = new Map();
+
+    const reaction_counts_and_user_ids = [...distinct_reactions.keys()].map((local_id) => {
+        const user_ids = user_map.get(local_id);
+        return {
+            count: user_ids.length,
+            user_ids,
+        };
+    });
+    const should_display_reactors = check_should_display_reactors(reaction_counts_and_user_ids);
+
     for (const local_id of distinct_reactions.keys()) {
         const reaction = distinct_reactions.get(local_id);
         const user_ids = user_map.get(local_id);
 
         message.clean_reactions.set(
             local_id,
-            make_clean_reaction({local_id, user_ids, ...reaction}),
+            make_clean_reaction({local_id, user_ids, should_display_reactors, ...reaction}),
         );
     }
 
-    // We do update_user_fields in a separate loop, because doing so
-    // lets us avoid duplicating check_should_display_reactors to
-    // determine whether to store in the vote_text field a count or
-    // the names of reactors (users who reacted).
-    const reaction_counts_and_user_ids = get_reaction_counts_and_user_ids(message);
-    const should_display_reactors = check_should_display_reactors(reaction_counts_and_user_ids);
-    for (const clean_reaction of message.clean_reactions.values()) {
-        update_user_fields(clean_reaction, should_display_reactors);
-    }
-
     // We don't maintain message.reactions when users react to
     // messages we already have a copy of, so it's safest to delete it
     // after we've processed the reactions data for a message into the
     // clean_reactions data structure, which we do maintain.
     delete message.reactions;
 }
 
-function make_clean_reaction({local_id, user_ids, emoji_name, emoji_code, reaction_type}) {
-    const clean_reaction_object = {
+function make_clean_reaction({
```

Comment:
I don't love this now duplicating all the logic in `update_user_fields`; that's just asking for trouble in a future edit that forgets to change both code paths. But we can probably address that via comments or a follow-up.

Reply:
I also wasn't a huge fan of that. Thanks for bringing it up, I made a shared `build_reaction_data` that both use.